crates/client/rpc/src/versions/v0_8_0/methods/read/get_messages_status.rs

```diff
@@ -0,0 +1,43 @@
+use crate::utils::ResultExt;
+use crate::versions::v0_7_1::methods::read::get_transaction_status::get_transaction_status;
+use crate::{Starknet, StarknetRpcApiError, StarknetRpcResult};
+use alloy::primitives::TxHash;
+use jsonrpsee::core::Serialize;
+use serde::Deserialize;
+use starknet_core::types::SequencerTransactionStatus;
+use starknet_types_core::felt::Felt;
+
+pub fn get_messages_status(starknet: &Starknet, transaction_hash: TxHash) -> StarknetRpcResult<Vec<MessageStatus>> {
+    let l1_handler_tx_hashes = starknet
+        .backend
+        .get_l1_handler_tx_hashes(transaction_hash)
+        .or_internal_server_error("Retrieving L1 handler transactions from database")?;
+    if l1_handler_tx_hashes.is_empty() {
+        return Err(StarknetRpcApiError::TxnHashNotFound);
+    }
+    let mut message_statuses = vec![];
+    for l1_handler_tx_hash in l1_handler_tx_hashes {
+        let finality_status = match get_transaction_status(starknet, l1_handler_tx_hash) {
```

Comment:
I think we will need to review the get_transaction_status function in sequencer mode, as we need to handle transactions in both the mempool and the pending state